Accounted for deleted users and companies in snapshot queries

alexey-rise opened the pull request:
@tejohnso please review.

tejohnso commented:
@alexey-rise Is a user supposed to be removed from *coreData.deleted* if they are no longer deleted?

With the snapshot now excluding deleted users, it seems we can have a situation where a user is deleted via apps, then recreated, and they'll never show up in the user snapshot because they are in *coreData.deleted*

alexey-rise commented:
@tejohnso No, nothing is ever removed from `coreData.deleted` table. That is an interesting problem caused by the fact that user ID is persistent (unlike company ID) and is not regenerated when a user is recreated because it's a email address and not a GUID.
To solve this problem we need to compare the timestamp of latest change in `coreData.users` and deletion timestamp in `coreData.deleted` for the given user ID. I will make the necessary changes in a new PR.
